python/graphstorm/gconstruct/construct_payload_graph.py

```diff
@@ -0,0 +1,592 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+import numpy as np
+import dgl
+import torch as th
+
+from .transform import parse_feat_ops, process_features, preprocess_features
+from .utils import update_two_phase_feat_ops
+from .payload_utils import (BaseApplicationError, MissingValError,
+                            InvalidFeatTypeError,
+                            DGLCreateError, MisMatchedTypeError,
+                            MissingColumnError, MisMatchedFeatureError)
+
+PAYLOAD_PROCESSING_STATUS = "status_code"
+PAYLOAD_PROCESSING_ERROR_CODE = "error_code"
+PAYLOAD_PROCESSING_RETURN_MSG = "message"
+PAYLOAD_GRAPH = "graph"
+PAYLOAD_GRAPH_NODE_MAPPING = "node_mapping"
+PROCESS_INDEX = 0
+
+
+def prepare_data(input_data, feat_ops):
+    """ Prepare phase one data for two-phase feature transformation.
+
+    The function parses a payload input that contains node/edge features.
+    The payload is parsed according to original gconstruct configuration
+    and transformation.
+
+    Parameters
+    ----------
+    input_data : dict
+        The json payload features input.
+    feat_ops : dict of FeatTransform
+        The operations run on the node features of the node file.
+
+    Returns
+    -------
+    dict : A dict of feature info.
+    """
+    if feat_ops is None:
+        raise MissingValError(value_name="feat_ops",
+                              input_name="prepare_data function call")
+    feat_info = preprocess_features(input_data, feat_ops)
+
+    # update_two_phase_feat_ops function expects an input dictionary indexed by integers
+    # e.g {chunk_index: chunk_data}
+    # Here we are only do single process feature processing,
+    # so we need to add a dummy index to the return dictionary
+    return {PROCESS_INDEX: feat_info}
+
+
+def get_conf(gconstruct_conf_list, type_name, structure_type):
+    """ Retrieve node/edge type gconstruct config. Will combine all feature configuration
+    for one node/edge type
+
+    For either GConstruct Node/Edge, feature transformation can be defined in multiple blocks,
```

Comment:
The rest of this comment seems like an implementation detail, is this an important user-facing function? Otherwise let's keep the doc minimal, this seems like a utility function that maybe shouldn't be public?

Reply:
It is not user-facing, but it is not straightforward for developers to understand the whole logic. Would recommend to leave the comments here for maintainence.